src/main/scala/com/lambtors/poker_api/module/poker/application/flop/FlopCardsAdder.scala

```diff
@@ -0,0 +1,37 @@
+package com.lambtors.poker_api.module.poker.application.flop
+
+import scala.concurrent.{ExecutionContext, Future}
+import com.lambtors.poker_api.module.poker.domain.{PlayerRepository, PokerGameRepository}
+import com.lambtors.poker_api.module.poker.domain.error.{FlopNotPossibleWhenItIsAlreadyGiven, PokerGameNotFound}
+import com.lambtors.poker_api.module.poker.domain.model.GameId
+import com.lambtors.poker_api.module.shared.domain.DeckProvider
+
+final class FlopCardsAdder(
+  repository: PokerGameRepository,
+  playerRepository: PlayerRepository,
+  deckProvider: DeckProvider
+)(implicit ec: ExecutionContext) {
+  def add(gameId: GameId): Future[Unit] = repository.search(gameId).map(
+    gameOpt =>
+      if (gameOpt.isEmpty) {
+        throw PokerGameNotFound(gameId)
```

Comment:
don't throw, return failed future

Reply:
geraaaaaard, imperative mode

when you throw an exception inside future, this is a future failed